File: cpp_programming/template/personInfo.cpp
#include "personInfo.h"
#include <iomanip>

using namespace std;

personInfo::personInfo() {
	this->_name = "";
	this->_phone = "";
	this->_address = "";
}
personInfo::personInfo(personInfo &info) {
	this->_name = info._name;
	this->_phone = info._phone;
	this->_address = info._address;
}
personInfo::~personInfo() {
	this->_name = "";
	this->_phone = "";
	this->_address = "";
}

string personInfo::name() {
	return this->_name;
}
void personInfo::name(string name) {
	this->_name = name;
}
string personInfo::phone() {
	return this->_phone;
}
void personInfo::phone(string phone) {
	this->_phone = phone;
}
string personInfo::address() {
	return this->_address;
}
void personInfo::address(string address) {
	this->_address = address;
}

personInfo& personInfo::operator= (personInfo& rinfo) {
	this->_name = rinfo._name;
	this->_phone = rinfo._phone;
	this->_address = rinfo._address;

	return *this;
}
bool personInfo::operator== (personInfo& rinfo) {
	bool isEqual = false;

	if (this->_name == rinfo._name) {
		isEqual = true;
	}

	return isEqual;
}
bool personInfo::operator== (string name) {
	bool isEqual = false;

	if (this->_name == name) {
		isEqual = true;
	}

	return isEqual;
}
ostream& operator<< (ostream &os, personInfo &rinfo) {
	os << left << setw(7) << rinfo.name() 
		<< left << setw(14) << rinfo.phone()
		<< left << setw(50) << rinfo.address();

	return os;
}

File: py_programming/personInfo.py
# -*- coding: utf-8 -*-

class personInfo(object):
    def __init__(self, info = None):
        self._name = info._name if info else ""
        self._phone = info._phone if info else ""
        self._address = info._address if info else ""

    def __del__(self):
        self._name = ""
        self._phone = ""
        self._address = ""

    @property
    def name(self):
        return self._name

    @name.setter
    def name(self, value):
        self._name = value

    @property
    def phone(self):
        return self._phone

    @phone.setter
    def phone(self, value):
        self._phone = value

    @property
    def address(self):
        return self._address

    @address.setter
    def address(self, value):
        self._address = value

    def __eq__(self, other):
        isEqual = False

        if type(other) is personInfo:
            if (self._name == other._name):
                isEqual = True
        elif type(other) is str:
            if (self._name == other):
                isEqual = True
        else:
            isEqual = False

        return isEqual
